File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp ;
        int n=nums.size();
        for(int i=0;i<n;i++){
            mp[nums[i]]=i;
        }
        vector<int> ans;
        for(int i=0;i<n;i++){
            int toFind=target-nums[i];
            if(mp.count(toFind) && mp[toFind]!=i){
                vector<int> temp;
                temp.push_back(i);
                temp.push_back(mp[toFind]);
                sort(temp.begin(),temp.end());
                return temp;
            }
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        mp.put(nums[0],0);
        for(int i=1;i<nums.length;i++){
            int required=target-nums[i];
            if(mp.get(required)!=null){
                return new int[]{mp.get(required),i};
            }
            mp.put(nums[i],i);
        }
        return new int[]{};
    }
}